cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/validator/privatedns/AzurePrivateDnsZoneValidatorService.java

```diff
@@ -0,0 +1,92 @@
+                    "name %s and try again.", existingPrivateDnsZoneId, serviceEnum.getResourceType(), serviceEnum.getDnsZoneName());
+            addValidationError(validationMessage, resultBuilder);
+        }
+        return resultBuilder;
+    }
+
+    public ValidationResult.ValidationResultBuilder privateDnsZoneExists(AzureClient azureClient, String privateDnsZoneId,
+            ValidationResult.ValidationResultBuilder resultBuilder) {
+        ResourceId privateDnsZoneResourceId = ResourceId.fromString(privateDnsZoneId);
+        boolean privateDnsZoneExists = azureClient.getPrivateDnsZonesByResourceGroup(privateDnsZoneResourceId.subscriptionId(),
+                        privateDnsZoneResourceId.resourceGroupName())
+                .stream()
+                .anyMatch(pz -> pz.name().equals(privateDnsZoneResourceId.name()));
+        if (!privateDnsZoneExists) {
+            String validationMessage = String.format("The provided private DNS zone %s does not exist. Please make sure the specified " +
+                    "private DNS zone exists and try environment creation again.", privateDnsZoneId);
+            addValidationError(validationMessage, resultBuilder);
+        }
+        return resultBuilder;
+    }
+
+    public ValidationResult.ValidationResultBuilder privateDnsZoneConnectedToNetwork(AzureClient azureClient, String networkResourceGroupName,
+            String networkName, String privateDnsZoneId, ValidationResult.ValidationResultBuilder resultBuilder) {
+        String networkId = azureClient.getNetworkByResourceGroup(networkResourceGroupName, networkName).id();
+        ResourceId privateDnsZoneResId = ResourceId.fromString(privateDnsZoneId);
+        PagedList<VirtualNetworkLinkInner> virtualNetworkLinks = azureClient.listNetworkLinksByPrivateDnsZoneName(privateDnsZoneResId.subscriptionId(),
+                privateDnsZoneResId.resourceGroupName(), privateDnsZoneResId.name());
+        List<String> connectedNetworks = virtualNetworkLinks.stream().map(vnl -> vnl.virtualNetwork().id()).collect(Collectors.toList());
+        if (!connectedNetworks.contains(networkId)) {
+            String validationMessage = String.format("The private DNS zone %s does not have a network link to network %s. Please make sure the private DNS " +
+                    "zone is connected to the network provided to the environment.", privateDnsZoneId, networkName);
+            addValidationError(validationMessage, resultBuilder);
+        }
+        return resultBuilder;
+    }
+
+    public ValidationResult privateDnsZonesNotConnectedToNetwork(AzureClient azureClient, String networkId, String resourceGroupName, String dnsZoneName,
+            ValidationResult.ValidationResultBuilder resultBuilder) {
+        PagedList<PrivateZone> privateDnsZoneList = azureClient.getPrivateDnsZoneList();
```

Comment:
this is a longer operation, should be out of the `for` loop, ideally

Reply:
have you missed this one or you don't agree, @gergopapi2?

Reply:
I missed it and yes, that is important. Changing it now. Thank you.